Add x media schema and tag schema

siman4457 opened the pull request:
## XMedia
`XMedia` stands for "Extended Media" and consists of any type of media that a user uploads with a Post. Currently we only support uploading photos with a Post. Eventually, XMedia should eventually replace the current `Media` collection.

### Workflow of creating a Post with XMedia (WIP):
1) User selects a photo, adds description and tag(s)
2) Photo gets uploaded to Sirv on the frontend
3) Get the response from Sirv
4) Trigger addPost API with Sirv response and user's input (description and tags)

### New Queries and Mutations
Queries:
- getXMedia

Mutations:
- addXMedia
- removeXMedia


## Tags
`Tags` link XMedia to climb(s) or area(s). XMedia will contain an array of tagIds.

### New Queries and Mutations
Queries:
- getTags

Mutations:
- addTag
- deleteTag

siman4457 committed:
Add addPost and removePost mutations.

siman4457 committed:
add getPost query. Add comments.

siman4457 committed:
Add XMedia schema and types. Add addXMedia, removeXMedia, getXMedia.

siman4457 committed:
Rename resolver file for XMedia.

siman4457 committed:
Add Tag Schema and Types. Add addTag, deleteTag, getTags.

siman4457 committed:
Merge branch 'develop' into addXMediaSchema

siman4457 committed:
Resolve linting error.

vnugent commented on src/db/XMediaSchema.ts:
```diff
@@ -0,0 +1,47 @@
+const { Schema } = mongoose
+
+const XMediaSchema = new Schema<XMediaType>({
+  userId: {
```
User id in Auth0 is in uuid v4 format (Ex: `96373086-2d70-4e92-8036-3947d4cec3f9`) so we'd need to use MUUID here.  See `mediaUuid`  below for declaration.

siman4457 replied:
I've made this change and also changed destinationId on XMedia Schema to also use MUUID since this is what Media Schema is using. Lastly, I added onModel to XMedia (also copied from Media Schema).

vnugent commented:
@siman4457 Looking great.  I requested a small change to userId then we're good to merge.

siman4457 committed:
Update Tag Schema and XMedia Schema.

Add onModel to Tag Schema. Change destinationId type to MUUID on Tag Schema. Change userId type to MUUID on XMedia Schema.

vnugent commented on src/graphql/posts/mutations.ts:
```diff
@@ -21,7 +21,6 @@ const PostMutations = {
   removePost: async (_: any, { input }: {input: RemovePostInputType}) => {
     const PostModel = getPostModel()
```
just in case you haven't planned to,  it's a good idea to put database logic in its own Data source classes so that the code in here is light.

siman4457 replied:
I'm planning to do that in the future. I wanted to start out by keeping it simple.
